sdk/storage/storage-blob/src/ServiceURL.ts

```diff
@@ -151,12 +164,85 @@ export class ServiceURL extends StorageURL {
     });
   }
 
+  /**
+   * Iterates over containers under the specified account.  The returned iterator is resumable.
+   *
+   * @example
+   * for await (const container of serviceURL.listContainers()) {
+   *   console.log(`Container: ${container.name}`);
+   * }
+   *
+   * @example
+   * let iter1 = serviceURL.listContainers();
+   * let i = 1;
+   * for await (const container of iter1) {
+   *   console.log(`${i}: ${container.name}`);
+   *   i++;
+   *   if (i > 5) {
+   *     break;
+   *   }
+   * }
+   *
+   * const iter2 = serviceURL.listContainers({ resumePoint: iter1.resumePoint });
+   * for await (const container of iter2) {
+   *   console.log(`${i}: ${container.name}`);
+   *   i++;
+   * }
+   *
+   * @param {IServiceListContainersSegmentOptions} [options]
+   * @returns {ResumableAsyncIterableIterator<Models.ContainerItem>}
+   * @memberof ServiceURL
+   */
+  public listContainers(
+    options?: IServiceListContainersSegmentOptions
+  ): ResumableAsyncIterableIterator<Models.ContainerItem> {
+    const serviceURL = this;
+    const resumePoint = !options || !options.resumePoint ? { } : options.resumePoint;
+    const aborter = !options || !options.abortSignal ? Aborter.none : options.abortSignal;
+    const iter: ResumableAsyncIterableIterator<Models.ContainerItem> =
+     (async function* items(): AsyncIterableIterator<Models.ContainerItem> {
+      do {
+        const listContainersResponse = await serviceURL.listContainersSegment(
+          resumePoint.nextMarker,
+          {
+            ... options,
+            abortSignal: aborter
+          });
+
+        const containers = listContainersResponse.containerItems;
+        for (let i = iter.resumePoint.lastIndex || 0; i < containers.length; i++) {
+          iter.resumePoint.lastIndex = i;
+          yield containers[i];
+        }
+
+        iter.resumePoint.nextMarker = listContainersResponse.nextMarker;
+        iter.resumePoint.lastIndex = 0;
+      } while (iter.resumePoint.nextMarker);
+    } as any)() as any;
+
+    iter.resumePoint = { ... resumePoint };
+    return iter;
+  }
+
+  public listAllContainers(options?: IServiceListContainersSegmentOptions) {
+    return {
+      [Symbol.asyncIterator]: () => this.listContainers(options),
+      then(res: any, rej: any) {
```

Comment:
Do we still want this for resolving all items?

Reply:
No.

Reply:
Should be removed everywhere. I like the idea in many ways but we decided not to expose get-all-into-an-array functionality. Users could use Ix or other async iterator libraries to quickly whip this up if they choose to.